system-addon/content-src/components/Card/Card.jsx

```diff
@@ -36,32 +37,116 @@ export class _Card extends React.PureComponent {
    */
   async maybeLoadImage() {
     // No need to load if it's already loaded or no image
-    const {image} = this.props.link;
-    if (!this.state.imageLoaded && image) {
+    const {cardImage} = this.state;
+    if (!cardImage) {
+      return;
+    }
+
+    const imageUrl = cardImage.url;
+    if (!this.state.imageLoaded) {
       // Initialize a promise to share a load across multiple card updates
-      if (!gImageLoading.has(image)) {
+      if (!gImageLoading.has(imageUrl)) {
         const loaderPromise = new Promise((resolve, reject) => {
           const loader = new Image();
           loader.addEventListener("load", resolve);
           loader.addEventListener("error", reject);
-          loader.src = image;
+          loader.src = imageUrl;
         });
 
         // Save and remove the promise only while it's pending
-        gImageLoading.set(image, loaderPromise);
-        loaderPromise.catch(ex => ex).then(() => gImageLoading.delete(image)).catch();
+        gImageLoading.set(imageUrl, loaderPromise);
+        loaderPromise.catch(ex => ex).then(() => gImageLoading.delete(imageUrl)).catch();
       }
 
       // Wait for the image whether just started loading or reused promise
-      await gImageLoading.get(image);
+      await gImageLoading.get(imageUrl);
 
       // Only update state if we're still waiting to load the original image
-      if (this.props.link.image === image && !this.state.imageLoaded) {
+      if (_Card.isImageInState(this.state, this.props.link.image) && !this.state.imageLoaded) {
         this.setState({imageLoaded: true});
       }
     }
   }
 
+  /**
+   * Checks if .image property on link object is a local image with blob data.
+   *
+   * @param {obj|string} image
+   * @returns {bool} true if image is a local image object, otherwise false
+   *                 (otherwise, image will be a URL as a string)
+   */
+  static isLocalImageObject(image) {
```

Comment:
Can you add a comment that is only works for props since state has `.url` and not `.data`

Reply:
Done!